external_search.py

```diff
@@ -268,375 +276,59 @@ def base_index_name(self, index_or_alias):
 
         return base_works_index
 
-    def query_works(self, library, query_string, media, languages, fiction, audiences,
-                    target_age, in_any_of_these_genres=[], on_any_of_these_lists=None, fields=None, size=30, offset=0):
+    def query_works(self, query_string, filter=None, pagination=None,
+                    debug=False):
+        """Run a search query.
+
+        :param query_string: The string to search for.
+        :param filter: A Filter object, used to filter out works that
+            would otherwise match the query string.
+        :param pagination: A Pagination object, used to get a subset
+            of the search results.
+        :param debug: If this is True, some debugging information will
+            be gathered (at a slight performance cost) and logged.
+        :return: A list of Work IDs that match the query string.
+        """
         if not self.works_alias:
             return []
 
-        if library is None:
-            # TODO: We're searching the entire index and filtering out
-            # books that aren't available to the patron's current
-            # library in a later step. To avoid search disruption,
-            # this is the behavior we'll use until we're able to
-            # migrate all existing search indexes to the new document
-            # format.
-            collection_ids = None
-        else:
-            collection_ids = [x.id for x in library.collections]
-
-        filter = self.make_filter(
-            collection_ids, media, languages, fiction,
-            audiences, target_age, in_any_of_these_genres,
-            on_any_of_these_lists
-        )
-        q = dict(
-            filtered=dict(
-                query=self.make_query(query_string),
-                filter=filter,
-            ),
-        )
-        body = dict(query=q)
-        search_args = dict(
-            index=self.works_alias,
-            body=dict(query=q),
-            from_=offset,
-            size=size,
-        )
-        if fields is not None:
-            search_args['fields'] = fields
-        # search_args['explain'] = True
-        # print "Args looks like: %r" % search_args
-        results = self.search(**search_args)
-        # print "Results: %r" % results
-        return results
-
-    def make_query(self, query_string):
```

Comment:
The major accomplishment of this branch is splitting up `make_query` into smaller, independently tested methods.